tests/conftest.py

```diff
@@ -22,27 +23,179 @@ def requires_module(function, name, call=None):
 requires_seaborn = partial(requires_module, name="seaborn")
 
 
-def generate_cov(n_trials, n_channels):
-    """Generate a set of cavariances matrices for test purpose"""
-    rs = np.random.RandomState(1234)
+def generate_cov(n_trials, n_channels, rs, return_params=False):
+    """Generate a set of covariances matrices for test purpose"""
     diags = 2.0 + 0.1 * rs.randn(n_trials, n_channels)
     A = 2 * rs.rand(n_channels, n_channels) - 1
     A /= np.linalg.norm(A, axis=1)[:, np.newaxis]
     covmats = np.empty((n_trials, n_channels, n_channels))
     for i in range(n_trials):
         covmats[i] = A @ np.diag(diags[i]) @ A.T
-    return covmats, diags, A
+    if return_params:
+        return covmats, diags, A
+    else:
+        return covmats
+
+
+@pytest.fixture
+def rndstate():
+    return np.random.RandomState(1234)
+
+
+@pytest.fixture
+def get_covmats(rndstate):
+    def _gen_cov(n_trials, n_chan):
+        return generate_cov(n_trials, n_chan, rndstate, return_params=False)
+
+    return _gen_cov
+
+
+@pytest.fixture
+def get_covmats_params(rndstate):
+    def _gen_cov_params(n_trials, n_chan):
+        return generate_cov(n_trials, n_chan, rndstate, return_params=True)
+
+    return _gen_cov_params
+
+
+@pytest.fixture
+def get_labels():
+    def _get_labels(n_trials, n_classes):
+        return np.arange(n_classes).repeat(n_trials // n_classes)
+
+    return _get_labels
+
+
+def is_positive_semi_definite(mats):
+    """Check if all matrices are positive semi-definite
+
+    Parameters
+    ----------
+    X : ndarray, shape (..., n, n)
+        The set of square matrices, at least 2D ndarray.
+
+    Returns
+    -------
+    ret : boolean
+        True if all matrices are positive semi-definite
+    """
+    cs = mats.shape[-1]
+    return np.all(np.linalg.eigvals(mats.reshape((-1, cs, cs))) >= 0.0)
+
+
+def is_positive_definite(mats):
+    """Check if all matrices are positive definite
+
+    Parameters
+    ----------
+    X : ndarray, shape (..., n, n)
+        The set of square matrices, at least 2D ndarray.
+
+    Returns
+    -------
+    ret : boolean
+        True if all matrices are positive definite.
+    """
+    cs = mats.shape[-1]
+    return np.all(np.linalg.eigvals(mats.reshape((-1, cs, cs))) > 0.0)
+
+
+def is_symmetric(X):
+    """Check if all matrices are symmetric.
+
+    Parameters
+    ----------
+    X : ndarray, shape (..., n, n)
+        The set of square matrices, at least 2D ndarray.
+
+    Returns
+    -------
+    ret : boolean
+        True if all matrices are symmetric.
+    """
+    # return assert_array_almost_equal(X, np.swapaxes(X, -2, -1), 6)
+    return X == approx(np.swapaxes(X, -2, -1))
 
 
 @pytest.fixture
-def covmats():
-    """Generate covariance matrices for test"""
-    covmats, _, _ = generate_cov(6, 3)
-    return covmats
+def is_spd():
```

Comment:
Naive question: what is the interest to use a fixture for this function, rather than using a classical function?

Reply:
It is more convenient to handle scope, but in this case functions are also ok.